src/components/AssetsPage/index.jsx

```diff
@@ -10,43 +10,98 @@ import { getAssets } from '../../data/actions/assets';
 import edxBootstrap from '../../SFE.scss';
 import styles from './AssetsPage.scss';
 
-class AssetsPage extends React.Component {
+export class AssetsPage extends React.Component {
   constructor(props) {
     super(props);
+
+    const hasAssets = this.hasAssets(props.assetsList);
+
     this.state = {
       checked: false,
+      hasAssets,
     };
   }
 
   componentDidMount() {
     this.props.getAssets(this.props.request, this.props.courseDetails);
   }
 
-  componentDidUpdate() {
-    this.props.getAssets(this.props.request, this.props.courseDetails);
+  componentWillReceiveProps(nextProps) {
```

Comment:
This handles changing hasAssets appropriately, based on the assetsList state. If the page hasAssets, and the nextProps have no assets, we switch to having no assets. If the page hasNoAssets, and the nextProps have assets, we switch to having assets.

assets + no content => no content
no content + new assets => new assets